### PR TITLE
Add typing for tmt/steps/provision/podman.py

### PR DESCRIPTION
**Pull request by janhavlin**

### CODE REVIEW

File: tmt/steps/provision/podman.py [hunk 1]
```diff
@@ -126,20 +133,24 @@ class GuestContainer(tmt.Guest):
...
         if self.opt('dry'):
             return
         # Check if the image is available
-        image_id = self.podman(
+        assert self.image is not None
+        podman_output = self.podman(
             ['images', '-q', self.image],
-            message=f"Check for container image '{self.image}'.")[0].strip()
+            message=f"Check for container image '{self.image}'.")
+        assert podman_output[0] is not None
```
Comment (thrix) [inline tmt/steps/provision/podman.py:152]: Can we rather return an error here, if the output is nothing, we should raise a resonable error ... ?

**Review by thrix — CHANGES_REQUESTED** [review]

File: tmt/steps/provision/podman.py (hunk 1 above)
Comment (janhavlin) [reply tmt/steps/provision/podman.py]: Okay, changed it to raising a `tmt.utils.RunError`.

**Review by thrix — APPROVED** [review]

File: tmt/steps/provision/podman.py [hunk 2]
```diff
@@ -67,16 +66,24 @@ def options(cls, how=None):
             click.option(
                 '-u', '--user', metavar='USER',
                 help='User to use for all container operations.')
-            ] + super().options(how)
+            ]
+        return options
+
+    @overload
+    def default(self, option: Literal['pull'], default: Any = None) -> bool: pass
```
Comment (happz) [inline tmt/steps/provision/podman.py:73]: I wouldn't bother, this needs a generic solution for all plugins :/

**Review by happz — APPROVED** [review]

File: tmt/steps/provision/podman.py (hunk 2 above)
Comment (janhavlin) [reply tmt/steps/provision/podman.py]: I got rid of the overloads. One test was failing due to importing of `Literal`. Not worth spending time to fix it.

**Pull request merged**